src/main/webapp/app/tasks/search-tasks/search-tasks.component.ts

```diff
@@ -0,0 +1,50 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { debounceTime, distinctUntilChanged, tap } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+
+@Component({
+  selector: 'tiny-search-tasks',
+  templateUrl: './search-tasks.component.html',
+  styleUrls: ['./search-tasks.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SearchTasksComponent implements OnInit {
+  @Input() set reset$(reset$: Subject<void>) {
+    reset$.subscribe(_ => {
+      this.resetSearch();
+    });
+  }
+  @Output() emitted = new EventEmitter<string>();
+  searchControl: FormControl;
+
+  constructor(private route: ActivatedRoute) { }
+
+  ngOnInit(): void {
+    this.searchControl = new FormControl('');
+    this.checkQueryFromUrl();
+    this.onValueChanged();
+  }
+
+  resetSearch(): void {
+    this.searchControl.reset();
+  }
+
+  private checkQueryFromUrl(): void {
+    this.route.queryParams.subscribe(params => {
+      const query = params.q;
+      if (query) {
+        this.searchControl.setValue(query);
+      }
+    });
+  }
+
+  private onValueChanged(): void {
+    this.searchControl.valueChanges.pipe(
+      debounceTime(300),
```

Comment:
this one is a bit tricky because when you open or reload the page with the q param set, it will first load the list will all items and then after 300ms do a sarch after the given value was set on the input field - how would you solve this better?

Reply:
I noticed it during debugging. It was just a quick decision. 
Definitely, it would be better to have the query params handler in app.component and  pass a query from URL to search component.

Reply:
But if you asked about `debounceTime`, it was used in order to decrease amount of requests (during typing a search query)